File: hoh/Jump _to_java/exception.cpp
#include <iostream>


class FoolException : std::exception {
};

class Sample {
	public:
		void sayNick(std::string nick) {
			if(nick == "fool") 
				throw FoolException();
			std::cout << "당신의 별명은 " << nick << "입니다" << std::endl;
    }
};


int main(void) {
	Sample sample = Sample();
	try {
		sample.sayNick("fool");
		sample.sayNick("genious");
	}
	catch(FoolException e) {
		std::cout << "FoolException이 발생했습니다." << std::endl;
	}
	
}

File: hoh/Jump _to_java/exception.java
class FoolException extends Exception {
}

class Sample {
    public void sayNick(String nick) throws FoolException{
            if("fool".equals(nick)) 
                throw new FoolException();
            System.out.println("당신의 별명은 "+nick+" 입니다.");
    }
}

public class exception {
	public static void main(String[] args) {
		Sample sample = new Sample();
		try {
			sample.sayNick("fool");
			sample.sayNick("genious");
		}
		catch(FoolException e) {
            System.err.println("FoolException이 발생했습니다.");
		}
        
	}
}
